### PR TITLE
[quant][core][gpu][cudnn] Added support for nhwc tensors in quantized cudnn add_relu op

### PR DESCRIPTION
**Pull request by dzdang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #75806

Summary:
When using the the quantized cudnn add operator, if the input tensors are 4D,
cudnn requires NHWC format in v. 8.4.0 (older versions may have relaxed this constraint).
Previously, all tensors defaulted to NCHW format.

Test plan:
```
python test/test_quantization.py -k test_qadd_relu_cudnn
```

Differential Revision: [D35651368](https://our.internmc.facebook.com/intern/diff/D35651368)

### CODE REVIEW

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35651368).

File: aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp [hunk 1]
```diff
@@ -73,18 +74,28 @@ Tensor add(Tensor qa, Tensor qb, double output_scale, int64_t output_zero_point)
...
   }
 
-  at::Tensor add_output = at::empty(qa.sizes(), at::device(at::kCUDA).dtype(at::kFloat));
-  at::Tensor quantized_output = at::_empty_affine_quantized(
-      qa.sizes(),
-      at::device(at::kCUDA).dtype(at::ScalarType::QInt8),
-      output_scale,
-      output_zero_point);
+  at::Tensor add_output = qa.dim() == 4 ? at::empty(qa.sizes(), at::device(at::kCUDA).dtype(at::kFloat), at::MemoryFormat::ChannelsLast)
+                                        : at::empty(qa.sizes(), at::device(at::kCUDA).dtype(at::kFloat));
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp:83]: nit: there are some duplicated code here, can we have a variable to represent memoryformat and only apply the branch for the variable? like:
```
memory_format = a if xxx else b
add_output = at:empty(qa.sizes(), ..., memory_format)
```

File: aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp [hunk 2]
```diff
@@ -73,18 +74,28 @@ Tensor add(Tensor qa, Tensor qb, double output_scale, int64_t output_zero_point)
...
+  at::Tensor quantized_output = qa.dim() == 4 ? at::_empty_affine_quantized(qa.sizes(),
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp:84]: same here

File: aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp [hunk 3]
```diff
@@ -73,18 +74,28 @@ Tensor add(Tensor qa, Tensor qb, double output_scale, int64_t output_zero_point)
...
+                                                                            output_scale,
+                                                                            output_zero_point,
+                                                                            at::MemoryFormat::ChannelsLast)
+                                              : at::_empty_affine_quantized(qa.sizes(),
+                                                                            at::device(at::kCUDA).dtype(at::ScalarType::QInt8),
+                                                                            output_scale,
+                                                                            output_zero_point);
   // TODO: When cudnn enables support for broadcasting, we can remove this tensor
-  at::Tensor requantize_multiplier_tensor = at::empty(quantized_output.sizes(), at::device(at::kCUDA).dtype(at::kFloat));
+  at::Tensor requantize_multiplier_tensor = qa.dim() == 4 ? at::empty(quantized_output.sizes(), at::device(at::kCUDA).dtype(at::kFloat), at::MemoryFormat::ChannelsLast)
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/cudnn/BinaryOps.cpp:94]: same for all the code in this change

**Review by jerryzh168 — CHANGES_REQUESTED** [review]
please remove duplicated code

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35651368).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35651368).

**Review by jerryzh168 — APPROVED** [review]

**Pull request closed without merging**